### PR TITLE
media-sound/pulseaudio-daemon: Untangle USE flags

### PR DESCRIPTION
**Pull request by i-garrison**
As suggested by @leio on irc

- Require USE `bluetooth` for codecs selected via USE `ldac` or `aptx`, and pull required `gstreamer` packages as build dependency but keep respective gstreamer plugins runtime dependency.
- Handle USE `gstreamer` to enable GStreamer-based RTP implementation only, and suggest `media-plugins/gst-plugins-opus` for RTP OPUS payload.
- Require USE `dbus` for USE `equalizer`, and pull required `sys-apps/dbus` as build and runtime dependency.
- Add USE `fftw` to enable `module-virtual-surround-sink`.

Additionally, native backend for HSP and HFP bluetooth profiles has no reported issues, therefore
- drop USE `native-headset` and enable native backend support for HSP and HFP bluetooth profiles unconditionally if USE `bluetooth`. This adds microphone support for most bluetooth headsets by default, and can be disabled by user via runtime configuration if needed (see added `elog`.)

### CODE REVIEW

**Comment by leio** [comment]
Second commit log talks about doing more than it actually does (talks about depending on dbus, but that seems already there and commit only edits metadata.xml about it).
Third commit log has a typo in the package name - you are changing pulseaudio, not pulseaudio-daemon, but have the latter at start of commit log - this made it very confusing when reading the first line of commit log body beyond summary.
I don't really know about the fourth commit about native-headset stuff and haven't really reviewed that one.

Other than that, looks good to me.

**Comment by i-garrison** [comment]
> Third commit log has a typo in the package name - you are changing pulseaudio, not pulseaudio-daemon, but have the latter at start of commit log - this made it very confusing when reading the first line of commit log body beyond summary.

Fixed commit message now.

**Comment by i-garrison** [comment]
> Second commit log talks about doing more than it actually does (talks about depending on dbus, but that seems already there and commit only edits metadata.xml about it).

RIght, amended that commit message to note what is actually changed. Guess that was just bad wording on my side, as I tried to describe how USE equalizer is implemented after the change (which is: it requres USE dbus and also pulls required sci-libs/fftw as build and runtime dependency.)

File: media-sound/pulseaudio-daemon/pulseaudio-daemon-16.0.ebuild [hunk 1]
```diff
@@ -78,14 +84,15 @@ COMMON_DEPEND="
 	equalizer? (
 		sci-libs/fftw:3.0
```
Comment (thesamesam) [inline media-sound/pulseaudio-daemon/pulseaudio-daemon-16.0.ebuild:85]: I think we probably need := on the FFTW deps (it exposes a subslot for the library it installs, it's the library we're using, and we link against it, so we should)

File: media-sound/pulseaudio-daemon/pulseaudio-daemon-16.0.ebuild [hunk 2]
```diff
@@ -126,12 +133,11 @@ RDEPEND="
...
 	bluetooth? (
-		gstreamer? (
-			ldac? ( media-plugins/gst-plugins-ldac )
-			aptx? ( media-plugins/gst-plugins-openaptx )
-		)
+		ldac? ( media-plugins/gst-plugins-ldac )
+		aptx? ( media-plugins/gst-plugins-openaptx )
 	)
 "
+unset gstreamer_deps
```
Comment (thesamesam) [inline media-sound/pulseaudio-daemon/pulseaudio-daemon-16.0.ebuild:140]: Nice.

**Review by thesamesam — APPROVED** [review]

**Pull request closed without merging**